src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -591,6 +591,23 @@ std::unique_ptr<LogIterator> FileBasedWal::iterator(LogID firstLogId,
     return std::make_unique<FileBasedWalIterator>(shared_from_this(), firstLogId, lastLogId);
 }
 
+bool FileBasedWal::linkCurrentWAL(const char* newPath) {
+    closeCurrFile();
+    std::lock_guard<std::mutex> g(walFilesMutex_);
+    if (walFiles_.empty()) {
+        LOG(INFO) << idStr_ << "Create link failed, there is no wal files!";
+        return false;
+    }
+    auto it = walFiles_.rbegin();
+    if (link(it->second->path(), newPath) != 0) {
+        LOG(INFO) << idStr_ << "Create link failed for " << it->second->path()
+                  << " on " << newPath << ", error:" << strerror(errno);
+        return false;
+    }
+    LOG(INFO) << idStr_ << "Create link success for " << it->second->path()
```

Comment:
I have a question, do we need a prepareNewFile after the hard link created?

Reply:
Because I'm worried that the link file will change with the source file.

Reply:
We don't need to call prepareNewFile here.  Because a new wal file will be created when next log appended.

Reply:
> Because I'm worried that the link file will change with the source file.

The WAL becomes immutable after closed.